staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/webhook_converter.go

```diff
@@ -153,6 +183,59 @@ func getRawExtensionObject(rx runtime.RawExtension) (runtime.Object, error) {
 	return &u, nil
 }
 
+// getConvertedObjectsFromResponse validates the response, and returns the converted objects.
+// if the response is malformed, an error is returned instead.
+// if the response does not indicate success, the error message is returned instead.
+func getConvertedObjectsFromResponse(expectedUID types.UID, response runtime.Object) (convertedObjects []runtime.RawExtension, err error) {
+	switch response := response.(type) {
+	case *v1.ConversionReview:
+		// Verify GVK to make sure we decoded what we intended to
+		v1GVK := v1.SchemeGroupVersion.WithKind("ConversionReview")
+		if response.GroupVersionKind() != v1GVK {
+			return nil, fmt.Errorf("expected webhook response of %v, got %v", v1GVK.String(), response.GroupVersionKind().String())
+		}
+
+		if response.Response == nil {
+			return nil, fmt.Errorf("no response provided")
+		}
+
+		// Verify UID to make sure this response was actually meant for the request we sent
+		if response.Response.UID != expectedUID {
+			return nil, fmt.Errorf("expected response.uid=%q, got %q", expectedUID, response.Response.UID)
+		}
+
+		if response.Response.Result.Status != metav1.StatusSuccess {
+			// TODO: Return a webhook specific error to be able to convert it to meta.Status
+			if len(response.Response.Result.Message) > 0 {
+				return nil, errors.New(response.Response.Result.Message)
+			}
+			return nil, fmt.Errorf("response.result.status was '%s', not 'Success'", response.Response.Result.Status)
+		}
+
+		return response.Response.ConvertedObjects, nil
+
+	case *v1beta1.ConversionReview:
+		// v1beta1 processing did not verify GVK or UID, so skip those for compatibility
```

Comment:
Thanks for adding a comment about preserving this behavior. I noticed it when e2e testing v1beta1 because our image previously didn't return apiVersion or kind.